src/backend/vulkan.c

```diff
@@ -152,18 +153,28 @@ VkPhysicalDevice get_physical_device(VkInstance inst) {
 		return VK_NULL_HANDLE;
 	}
 
-/*	struct VkPhysicalDevicePCIBusInfoPropertiesEXT busProps;
-	struct VkPhysicalDeviceProperties2 aa = {
+	VkPhysicalDeviceDrmPropertiesEXT drm_props = {
+    		.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_DRM_PROPERTIES_EXT,
+	};
+
+	VkPhysicalDeviceProperties2 dev_props = {
 		.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PROPERTIES_2,
-		.pNext = &busProps,
-	}; XXX broken
-	vkGetPhysicalDeviceProperties22(pdev, &aa);*/
+		.pNext = &drm_props,
+	};
+	vkGetPhysicalDeviceProperties2(pdev, &dev_props);
 	VkPhysicalDeviceProperties props;
 	vkGetPhysicalDeviceProperties(pdev, &props);
 	boxlog("Device name: %s", props.deviceName);
 	boxlog("       type: %s", type_to_string(props.deviceType));
-/*	errlog("%d %d %d %d", busProps.pciDomain, busProps.pciBus,
-	 busProps.pciDevice, busProps.pciFunction);*/
+	if (drm_props.hasPrimary) {
+		boxlog("    primary: %d:%d", drm_props.primaryMajor, drm_props.primaryMinor);
+		*major = drm_props.primaryMajor;
+		*minor = drm_props.primaryMinor;
+	} else {
+    		errlog("failed to get primary device");
```

Comment:
If there is no primary node associated to the device, we should abort (we can't do modesetting). Let's add `return VK_NULL_HANDLE;` here.
Later we need to add some proper logic for the device selection, right now it is broken if there is more than 1 gpu.

Reply:
Done that. Unfortunately my only dual GPU system has an nvidia card :confused: